src/System.Security.Cryptography.Algorithms/src/Internal/Cryptography/AesImplementation.Unix.cs

```diff
@@ -1,34 +1,74 @@
 ﻿// Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+using System;
 using System.Security.Cryptography;
 
 namespace Internal.Cryptography
 {
     internal partial class AesImplementation
     {
-        private static ICryptoTransform CreateEncryptor(
+        private static ICryptoTransform CreateTransformCore(
             CipherMode cipherMode,
             PaddingMode paddingMode,
             byte[] key,
             byte[] iv,
-            int blockSize)
+            int blockSize,
+            bool encrypting)
         {
-            return new AesOpenSslCryptoTransform(cipherMode, paddingMode, key, iv, blockSize, true);
-        }
+            Func<IntPtr> algorithmFunc = FindAlgorithmSelector(key.Length * 8, cipherMode);
 
-        private static ICryptoTransform CreateDecryptor(
-            CipherMode cipherMode,
-            PaddingMode paddingMode,
-            byte[] key,
-            byte[] iv,
-            int blockSize)
-        {
-            return new AesOpenSslCryptoTransform(cipherMode, paddingMode, key, iv, blockSize, false);
+            // The algorithm pointer is a static pointer, so not having any cleanup code is correct.
+            IntPtr algorithm = algorithmFunc();
```

Comment:
Is there any other caller of FindAlgorithmSelector that needs to do something other than invoke it?  I'm wondering if FindAlgorithmSelector should just be SelectAlgorithm and effectively include this delegate invocation.